docs/manual/index-checker.tex

```diff
@@ -457,8 +460,65 @@ \section{Binary search indices\label{index-searchindex}}
   This is the top type, and programmers should rarely need to write it.
 \end{description}
 
+\section{IndexOf indices\label{index-indexof}}
+
+The sole purpose of the IndexOf Checker is to allow expressing the
```

Comment:
This mention of "the IndexOf Checker" (an implementation detail) comes out of the blue.  What is it, and why should the user care?  The manual should start with things that the user cares about, in terms that the user can understand.